Separate manager page to manage deleted resources #modxbughunt

wuuti opened the pull request:
### What does it do?
Adds a _Trash_ menu entry and a Trash managing page. The page lists all deleted resources together with their contexts, pagetitle, longtitle, the deletion date and the username who deleted the resource.

<img width="1001" alt="screen shot 2017-07-07 at 15 33 56" src="https://user-images.githubusercontent.com/2852461/27960001-ea586064-6329-11e7-8169-472d22d269c8.png">
<img width="759" alt="screen shot 2017-07-07 at 15 34 22" src="https://user-images.githubusercontent.com/2852461/27960003-eda978e8-6329-11e7-9a11-23243b037381.png">
<img width="757" alt="screen shot 2017-07-07 at 15 34 43" src="https://user-images.githubusercontent.com/2852461/27960005-ef68c68e-6329-11e7-9054-fad259fc7787.png">

Resources can be restored or purged by the context menu, by the bulk actions or all resources can be purged with a button. The button fullfils the same functionality as the bin icon above the resource tree in previous versions. 
With this merge the behaviour of the bin icon changes, it now directs you to the trash manager page. 

### Why is it needed?
The former behaviour was an all or nothing purge. It was not possible, for example, to purge only resources from a certain context. This can now be easily done. Also the restoring of resources was not possible for multiple resources at once. And if deleted resources where hidden from the tree (e.g. inside a collections container), users may completely overlook them. 

### Related issue(s)/PR(s)
Solves parts of the multi-issue #764.

OptimusCrime committed:
Add property to modResource class

OptimusCrime committed:
Add property to modResource class

OptimusCrime committed:
Undo accidental changes to modResource

OptimusCrime committed:
Undo accidental changes to modResource

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added first version of trash CMP with ExtJs base.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added translations.
Added output of resources contexts.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Added purge function.
Added undelete call (restore).
Added specific confirmation message.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Implemented bulk purge.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Show list of resources in confirmation dialogues.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
[ReUp] Added trash to menus.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

Commit:
Added lexicon string for deleteby user column.

opengeek committed:
Change new installs to create tables with InnoDB engine

opengeek committed:
Change new installs to create tables with InnoDB engine

opengeek committed:
Conditionally override engine to MyISAM if MySQL < 5.6

opengeek committed:
Conditionally override engine to MyISAM if MySQL < 5.6

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

wuuti committed:
Added tree refresh on purge/restore in the grid.
Added path breadcrumb and ids to confirmation dialogue.

lawrenz1337 committed:
fix(Resource) delete dialog UI

lawrenz1337 committed:
fix(Resource) delete dialog UI

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Added tree and panel refreshes on actions.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

wuuti committed:
Added lexicon entries.
Changed recycle bin icon action to open the trash manager.
Changed recycle bin icon tooltip.
Added autosave to trash grid.

rtripault commented:
This looks promising!

A few notes : 

* do we want to delete the existing "purge all" button (i see it handy for single context setups)
* i would avoid using a tab panel if we do not intend to ship extra "tabs"

lawrenz1337 commented:
That is really cool. Great job! I'm so excited to try this out

wuuti commented:
@rtripault I am not an expert in extjs unfortunately. That whole part was copied and adjusted from sources (where, you are right, 2 tabs are used). So I would be happy if anyone else could give me a hint how to tweak it not to use a panel.

rtripault commented:
Happy to do so with a full review!

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+     */
+    public function getSortClassKey() {
+        return 'modResource';
+    }
```
This is not needed as the parent behavior is to check against `$this->classKey`

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+class modResourceTrashGetListProcessor extends modObjectGetListProcessor {
+
+    public $classKey = 'modResource';
+    public $languageTopics = [ 'resource' ];
```
Short array syntax, so PHP 5.4+ (yeah... i know...)

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+            );
+        }*/
+        return $list;
+    }
```
This does nothing more than the parent class

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+        // but we need the purge_deleted permission - for every single file
+
+        if (!empty($query)) {
+            $c->where([ 'modResource.pagetitle:LIKE' => '%' . $query . '%' ]);
```
Short array syntax again

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+
+        if (!empty($query)) {
+            $c->where([ 'modResource.pagetitle:LIKE' => '%' . $query . '%' ]);
+            $c->orCondition([ 'modResource.longtitle:LIKE' => '%' . $query . '%' ]);
```
Short array syntax again

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+        }
+        $c->where([
+            'modResource.deleted' => true,
+        ]);
```
Short array syntax again

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+        // parent-child path to the resource
+
+
+        $parents = [];
```
Short array syntax again

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+//        $canEdit = $this->modx->hasPermission('');
+//       $canSave = $this->modx->hasPermission('source_save');
+//        $canRemove = $this->modx->hasPermission('source_delete');
+        $canView = $this->modx->hasPermission('view_document');
```
This does not appear to be used

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+//        $objectArray = $object->toArray();
+        $objectArray['iconCls'] = $this->modx->getOption('mgr_source_icon', null, 'icon-folder-open-o');
+
+        $cls = [];
```
Short array syntax again

rtripault commented on core/model/modx/processors/resource/trash/purge.class.php:
```diff
@@ -0,0 +1,111 @@
+    public function process() {
+        /* get resources */
+        $id = $this->getProperty( 'id', false );
+        if ( empty( $id ) ) return $this->modx->lexicon( 'resource_err_ns' );
```
This appears to be redundant with what's already done in `initialize`

rtripault commented on core/model/modx/processors/resource/trash/purge.class.php:
```diff
@@ -0,0 +1,111 @@
+                'id:IN' => explode(',',$id ))
+        );
+        $count = count( $resources );
+        $this->modx->log( 1, "Resources found: " . $count );
```
Please use `MODX_LOG_LEVEL_ERROR` instead (used everywhere else in code base + easier to read)

rtripault commented on core/model/modx/processors/resource/trash/purge.class.php:
```diff
@@ -0,0 +1,111 @@
+        /** @var modResource $resource */
+        foreach ( $resources as $resource ) {
+            $ids[] = $resource->get( 'id' );
+        }
```
Here you appear to collect all resource IDs while you apparently can get them using the `explode` you do a bit before. Is that to make sure you only get valid IDs ?

rtripault commented on core/model/modx/processors/resource/trash/purge.class.php:
```diff
@@ -0,0 +1,111 @@
+            /** @var modTemplateVarResource $templateVarResource */
+            foreach ( $templateVarResources as $templateVarResource ) {
+                $templateVarResource->remove();
+            }
```
While this exists in current `modResourceEmptyRecycleBinProcessor`, i'm wondering if 

* that's required (schema relationships should handle this i guess)
* it should not occur after the `$resource->remove()` method (to make sure it went fine)

wuuti replied:
@rtripault I am not sure this is handled automatically. Will test this. Regarding the order removal, you mean we should first remove the resource, and only if that was fine, then remove the template vars? would be the right order, but how do we get the template vars once the resource is deleted?

rtripault commented on manager/assets/modext/widgets/resource/modx.panel.trash.js:
```diff
@@ -0,0 +1,359 @@
+            , getState: function () {
+                return {activeTab: this.items.indexOf(this.getActiveTab())};
+            }
+        })]
```
If you do not want the tab panel, just use your form panel : replace the whole `MODx.getPageStructure(` with

```
{
            layout: 'form'
            , title: _('trash.tab_title')
            , items: [{
                html: '<p>' + _('trash.intro_msg') + '</p>'
                , xtype: 'modx-description'
            }, {
                xtype: 'modx-grid-trash'
                , id: 'modx-trash-resourcelist'
                , cls: 'main-wrapper'
                , preventRender: true
            }]
        }
```

sonicpunk commented:
#modxbughunt #3points to @wuuti

rtripault commented on manager/controllers/default/resource/trash/index.class.php:
```diff
@@ -0,0 +1,53 @@
+     */
+    public function getTemplateFile() {
+        return ''; //'resource/trash/index.tpl';
+    }
```
This is already what the parent class does, so no need to override

rtripault commented on manager/controllers/default/resource/trash/index.class.php:
```diff
@@ -0,0 +1,53 @@
+     * @return bool
+     */
+    public function checkPermissions() {
+        return $this->modx->hasPermission('source_view');
```
I did not find any reference to this permission, also shouldn't it be `purge_deleted` ?

wuuti replied:
It should be 'menu_trash' here. Even if you don't have purge_deleted, you should be able to visit the trash page then and view+restore files.

rtripault commented on manager/controllers/default/resource/trash/index.class.php:
```diff
@@ -0,0 +1,53 @@
+     */
+    public function process(array $scriptProperties = []) {
+        $this->modx->lexicon->load('core:trash');
+    }
```
Why loading the lexicon strings here while it's supposed to be already done in `getLanguageTopics()`

rtripault commented on manager/templates/default/resource/trash/index.tpl:
```diff
@@ -0,0 +1,52 @@
+            </em>
+        </span>
+    </div>
+</div>
```
Is this file used ?

wuuti replied:
No.

rtripault commented on _build/data/transport.core.menus.php:
```diff
@@ -226,10 +226,22 @@
+    'text' => 'trash_menu',
+    'description' => 'trash_menu_desc',
+    'parent' => 'manage',
+    'permissions' => '',
```
It would probably be better to add some permission here, to make sure the menu is only displayed to users having the requires perm(s)

wuuti replied:
Right. It was supposed to be "menu_trash", just forgot to add that to the build file.

rtripault commented on core/model/modx/processors/resource/trash/getlist.class.php:
```diff
@@ -0,0 +1,130 @@
+        $objectArray['iconCls'] = $this->modx->getOption('mgr_source_icon', null, 'icon-folder-open-o');
+
+        $cls = [];
+        $cls[] = 'restore';
```
You might want to check if the user has `undelete_document` permission

rtripault commented on core/model/modx/processors/resource/trash/purge.class.php:
```diff
@@ -0,0 +1,111 @@
+        if ( empty( $this->id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $this->resource = $this->modx->getObject( 'modResource', $this->id );
+        if ( empty( $this->resource ) ) return $this->modx->lexicon( 'resource_err_nfs', array( 'id' => $this->id ) );
```
If you pass a comma separated list here, this should not work, preventing "batch purging"

rtripault reviewed (commented):
Added some comments (i now need to step back a bit ;p)

gpsietzema commented:
#modxbughunt #1point to @rtripault

Commit:
Fix issue #12430 #modxbughunt

Commit:
Fix issue #12430 #modxbughunt

OptimusCrime commented:
Holy sh!! Nice work! :D

wuuti commented:
@rtripault good points, I will fix what you mentioned (not today any more)...

Jako committed:
Secondary button height fixes

Jako committed:
Secondary button height fixes

opengeek committed:
Fix collapsing secondary buttons

opengeek committed:
Fix collapsing secondary buttons

sdrenth commented:
@wuuti Nice work, I think this would be a really nice addition! 

After testing the PR, this is my feedback.
- After I ran the build and setup, It's showing the lexicon keys in the menu instead of the values. 
![screen shot 2017-08-31 at 09 53 50](https://user-images.githubusercontent.com/7802465/29912555-54dba904-8e32-11e7-88d8-772656bc5c9b.png)
- If I delete a single resource which contains child resources, the child resources are deleted as well. Maybe add a check if resource has children then add an additional warning in the popup or generate a list of all the resources children that will also be deleted. But then a list of all nested child resources that would be deleted as well  should be generated.
- Generated popup when clicking restore/purge from grid is not wide enough to contain all content (first time the popup is generated shows correct width, next times it is too small)

![screen shot 2017-08-31 at 09 51 08](https://user-images.githubusercontent.com/7802465/29912525-3e4dee36-8e32-11e7-9360-2fabb9ea4571.png)
- Bulk action "Restore selected resources" generates an "action_err_ns" error because the connector returns a 404.
- Clicking the trash can in the context menu shows a tooltip but nothing else is happening. I would expect that clicking the trash can would result in opening the trash manager page. 
- After clicking "purge everything" refresh the resource tree?

OptimusCrime commented:
@wuuti Are you going to fix these things, or can someone else fork your fork and continue work on it?

Mark-H committed:
Added plugins and snippets to array to strip php tags from

(cherry picked from commit c876512f5cfe8c589803b3e3d51c89ccc0f5a624)

Mark-H committed:
Added plugins and snippets to array to strip php tags from

(cherry picked from commit c876512f5cfe8c589803b3e3d51c89ccc0f5a624)

wuuti commented:
@OptimusCrime I already started fixing some of the issues, but I would welcome anyone else supporting me, esp. with the extjs related issues.

wuuti commented:
@sdrenth Thanks for your review. Some remarks:
* I also see the problems with the lexicon entries sometimes, but currently do not know why. Maybe caching, working on it, but help appreciated.
* Regarding the deletion of child resources (did you mean "purging" of child resources?): good point. If a parent resource is purged, what happens to the child resources? At least a warning is needed, but also I see 2 different ways to proceed: purge the childs as well or rebase them to the "next" parent. Research and discussion needed
* bulk restoring does not work: confirmed. Will fix.
* cut of extjs popups: confirmed - but way beyond my extjs knowledge right now. Help appreciated.
* clicking the trash can works in my test environments as expected, you get to the trash manager page. Is there anything in the js console for you?
* after clicking the purge everything refresh tree: good catch, will implement this as well.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Restored PHP compatibility with arrays.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed accidently added example code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed redundant code.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Removed debug message.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Added missing permission for menu.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Removed unnecessary code and fixed permission.
Deleted (currently) unused template file.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

Commit:
Added tree and trash icon refreshes on actions.

wuuti commented:
Still in progress/open todo: 
* bulk restore
* extjs problems
* handling of child resources on purging/restoring

Commit:
Added restore processor.

Commit:
Added restore processor.

Commit:
Added restore processor.

Commit:
Added restore processor.

opengeek committed:
Handle version detection appropriately for MariaDB

opengeek committed:
Handle version detection appropriately for MariaDB

Commit:
Added lexicon entries.

Commit:
Added lexicon entries.

Commit:
Added lexicon entries.

Commit:
Added lexicon entries.

Commit:
Added refresh function.

Commit:
Added refresh function.

Commit:
Added refresh function.

Commit:
Added refresh function.

opengeek committed:
Enable rendering of SVG previews in Media Browser

opengeek committed:
Additional SVG preview improvements

- Add SVG support for IE and Edge
- Add proportional SVG thumbnail scaling
- Add SVG support to phpthumb for S3
- Check background for transparent images

opengeek committed:
Additional SVG preview improvements

- Add SVG support for IE and Edge
- Add proportional SVG thumbnail scaling
- Add SVG support to phpthumb for S3
- Check background for transparent images

Jako committed:
Fix javascript issue on resources containing a quote sign
- Introduced in #13497

Jako committed:
Fix javascript issue on resources containing a quote sign
- Introduced in #13497

Commit:
Reset image_width and image_height, remove introduced file_width and file_height

Commit:
Reset image_width and image_height, remove introduced file_width and file_height

Commit:
change modifier name

Commit:
change modifier name

Commit:
Revert "change modifier name"

This reverts commit 7bd1a2a89b9706418e945809aae059dc7bf51f2a.

Commit:
Revert "change modifier name"

This reverts commit 7bd1a2a89b9706418e945809aae059dc7bf51f2a.

rainbowtiger commented:
Really looking forward to having this available. It's been needed for a long time.

I vote for having an option to retain children, moving them up to the next parent.

Commit:
Current state.

Commit:
Current state.

Commit:
Current state.

Commit:
Current state.

Jako commented:
OMG, it seems that you have 'enhanced' your PR a lot. It could i.e. happened by merging the latest MODX changes into the trash branch - never do that …

wuuti commented:
@Jako yeah, I puzzled everything there while looking forward to the snowup :-(. Will check that later.

Mark-H commented:
If you need some git-fu help, feel free to ask me during the snowup to take a look :)

Commit:
Current state.

Commit:
Current state.

Commit:
Merge remote-tracking branch 'origin/trash' into trash

# Conflicts:
#	manager/assets/modext/widgets/resource/modx.panel.trash.js

wuuti commented:
Instead of trying to fix my broken branch/PR I cherry-picked the files and created a fresh branch on top of 2.x. Sorry for the weirdness here. Closing this in favor of the new one.
